src/appleseed/renderer/modeling/scene/visibilityflags.h

```diff
@@ -62,6 +62,7 @@ class APPLESEED_DLLSYMBOL VisibilityFlags
         GlossyRay       = 1 << 6,
         SpecularRay     = 1 << 7,
         SubsurfaceRay   = 1 << 8,
+        NPRRay          = 1 << 9,
```

Comment:
Out of curiosity, why is a new ray type necessary?

Reply:
NPR shaders are executed twice. The first time they work as normal OSL shaders and you can use any shading closure.
The second time it is executed the ray type is NPR. You can fetch the result of shading using getattribute, modify or replace it and pass the final result to NPR closures.

Reply:
Ok, thanks.